app/utils/get_filtered_records.py

```diff
@@ -144,33 +144,22 @@ def get_filtered_records(research_fields=False, filters=None, columns=None,
     if query_dicts is None or len(query_dicts) == 0:
         return []
 
-    result = []
-
     # Return all records if no filters are passed in
+    result = query_dicts
+
     if filters:
-        def should_include(d, k, v):
-            if len(v) == 0:
+        def should_be_included(record, field, permitted_values):
```

Comment:
don't see a clear reason as to why we defined these functions as closures. let's move this and the `date_filter` function outside of `get_filtered_records `

Reply:
And we can add typing while we are at it

Reply:
agreed, move outside and add typing